src/commands/force/source/open.ts

```diff
@@ -98,18 +101,27 @@ export class Open extends SourceCommand {
     return this.flags.urlonly ? result : this.openBrowser(url, result);
   }
 
-  private async setUpOpenPath(): Promise<string> {
+  private async setUpOpenPath(pageType: string): Promise<string> {
     try {
-      const flexipage = await this.org
-        .getConnection()
-        .singleRecordQuery<{ Id: string }>(
-          `SELECT id FROM flexipage WHERE DeveloperName='${path.basename(
-            this.flags.sourcefile as string,
-            '.flexipage-meta.xml'
-          )}'`,
-          { tooling: true }
-        );
-      return `/visualEditor/appBuilder.app?pageId=${flexipage.Id}`;
+      if (pageType === 'FlexiPage') {
+        const flexipage = await this.org
+          .getConnection()
+          .singleRecordQuery<{ Id: string }>(
+            `SELECT id FROM flexipage WHERE DeveloperName='${path.basename(
+              this.flags.sourcefile as string,
+              '.flexipage-meta.xml'
+            )}'`,
+            { tooling: true }
+          );
+        return `/visualEditor/appBuilder.app?pageId=${flexipage.Id}`;
+      } else if (pageType === 'ApexPage') {
+        return `/apex/${path
+          .basename(this.flags.sourcefile as string)
+          .replace('.page-meta.xml', '')
+          .replace('.page', '')}`;
```

Comment:
``` suggestion
          .replace(/\.page-meta\.xml|\.page/, '')}`;
```

Reply:
🤷🏻‍♂️   that's a close one.  I think I'll leave as-is for readability, since perf is negligible here.